codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttEncoder.java

```diff
@@ -358,6 +535,94 @@ private static ByteBuf encodeMessageWithOnlySingleByteFixedHeader(
+                case TOPIC_ALIAS_MAXIMUM:
+                case TOPIC_ALIAS:
+                    writeVariableLengthInt(propertiesBuf, property.propertyId);
+                    final short twoBytesInPropValue = ((MqttProperties.IntegerProperty) property).value.shortValue();
+                    propertiesBuf.writeShort(twoBytesInPropValue);
+                    break;
+                case PUBLICATION_EXPIRY_INTERVAL:
+                case SESSION_EXPIRY_INTERVAL:
+                case WILL_DELAY_INTERVAL:
+                case MAXIMUM_PACKET_SIZE:
+                    writeVariableLengthInt(propertiesBuf, property.propertyId);
+                    final int fourBytesIntPropValue = ((MqttProperties.IntegerProperty) property).value;
+                    propertiesBuf.writeInt(fourBytesIntPropValue);
+                    break;
+                case SUBSCRIPTION_IDENTIFIER:
+                    writeVariableLengthInt(propertiesBuf, property.propertyId);
+                    final int vbi = ((MqttProperties.IntegerProperty) property).value;
+                    writeVariableLengthInt(propertiesBuf, vbi);
+                    break;
+                case CONTENT_TYPE:
+                case RESPONSE_TOPIC:
+                case ASSIGNED_CLIENT_IDENTIFIER:
+                case AUTHENTICATION_METHOD:
+                case RESPONSE_INFORMATION:
+                case SERVER_REFERENCE:
+                case REASON_STRING:
+                    writeVariableLengthInt(propertiesBuf, property.propertyId);
+                    writeUTF8String(propertiesBuf, ((MqttProperties.StringProperty) property).value);
+                    break;
+                case USER_PROPERTY:
+                    final List<MqttProperties.StringPair> pairs = ((MqttProperties.UserProperties) property).value;
+                    for (MqttProperties.StringPair pair: pairs) {
+                        writeVariableLengthInt(propertiesBuf, property.propertyId);
+                        writeUTF8String(propertiesBuf, pair.key);
+                        writeUTF8String(propertiesBuf, pair.value);
+                    }
+                    break;
+                case CORRELATION_DATA:
+                case AUTHENTICATION_DATA:
+                    writeVariableLengthInt(propertiesBuf, property.propertyId);
+                    final byte[] binaryPropValue = ((MqttProperties.BinaryProperty) property).value;
+                    propertiesBuf.writeShort(binaryPropValue.length);
+                    propertiesBuf.writeBytes(binaryPropValue, 0, binaryPropValue.length);
+                    break;
+                default:
+                    //shouldn't reach here
+                    throw new EncoderException("Unknown property type: " + propertyType);
+            }
+        }
+        writeVariableLengthInt(propertiesHeaderBuf, propertiesBuf.readableBytes());
+        propertiesHeaderBuf.writeBytes(propertiesBuf);
+        propertiesBuf.release();
```

Comment:
I wonder if we should better do the release in a finally block. Just in case if something goes wrong... otherwise we will leak memory... This will also cover the `default:` case then